pkg/rpc/server/listen_test.go

```diff
@@ -104,6 +106,15 @@ func Test_dbWorker_update(t *testing.T) {
 			args:    args{appVersion: "1"},
 			wantErr: "failed DB hot update",
 		},
+		{
+			name: "Nil GaugeVec returns an error",
+			needsUpdate: needsUpdate{
+				input:  needsUpdateInput{appVersion: "1", skip: false},
+				output: needsUpdateOutput{needsUpdate: true},
+			},
+			args:    args{appVersion: "1", nilGaugeVec: true},
+			wantErr: "Prometheus gauge found to be nil: Prometheus gauge found to be nil",
```

Comment:
I think you're fine with just having `Prometheus gauge found to be nil` as `wantErr`.